engine/gamesys/src/gamesys/components/comp_sprite.cpp

```diff
@@ -1135,32 +1149,28 @@ namespace dmGameSystem
 
             if (!CanUseQuads(&textures))
             {
-                ResolveUVDataFromGeometry(&textures, scratch_uvs, component->m_FlipHorizontal, component->m_FlipVertical);
-
-                // Depending on the sprite is flipped or not, we loop the vertices forward or backward
-                // to respect face winding (and backface culling)
                 const dmGameSystemDDF::TextureSetAnimation* animation_ddf = textures.m_Animations[0];
+
                 int flipx = animation_ddf->m_FlipHorizontal ^ component->m_FlipHorizontal;
                 int flipy = animation_ddf->m_FlipVertical ^ component->m_FlipVertical;
+                float scaleX = flipx ? -1 : 1;
+                float scaleY = flipy ? -1 : 1;
+
+                // Depending on the sprite is flipped or not, we loop the vertices forward or backward
+                // to respect face winding (and backface culling)
                 int reverse = flipx ^ flipy;
 
+                ResolvePositionAndUVDataFromGeometry(&textures, &scratch_pos, scratch_uvs, scaleX, scaleY, reverse);
+
                 const Matrix4& w = component->m_World;
                 const dmGameSystemDDF::SpriteGeometry* geometry = textures.m_Geometries[0];
-                uint32_t num_points = geometry->m_Vertices.m_Count / 2;
-                const float* points = geometry->m_Vertices.m_Data;
-                const float* uvs    = geometry->m_Uvs.m_Data;
 
-                float scaleX = flipx ? -1 : 1;
-                float scaleY = flipy ? -1 : 1;
-
-                int step = reverse ? -2 : 2;
-                points = reverse ? points + num_points*2 - 2 : points;
-                uvs = reverse ? uvs + num_points*2 - 2 : uvs;
-
-                for (uint32_t vert = 0; vert < num_points; ++vert, points += step, uvs += step)
+                uint32_t num_points = scratch_pos.Size() / 2;
+                for (uint32_t vert = 0; vert < num_points; ++vert)
```

Comment:
Now the array is already in the correct order, so we can simply iterate through it.